File: commerceindex-sdk/python/src/commerceindex/scoring.py
"""CommerceIndex SDK — CI Score operations."""

from __future__ import annotations

from typing import TYPE_CHECKING

from commerceindex.models import CIScore

if TYPE_CHECKING:
    from commerceindex.client import CommerceIndex


class ScoreClient:
    """CI Score operations."""

    def __init__(self, client: CommerceIndex):
        self._client = client

    async def get(self, agent_id: str) -> CIScore:
        """Get full CI Agent Score breakdown for an agent."""
        data = await self._client._request("GET", f"/v1/agents/{agent_id}/score")
        return CIScore(**data)

File: commerceindex-sdk/typescript/src/scoring.ts
/** CommerceIndex SDK — CI Score operations. */

import type { CommerceIndex } from "./client.js";
import type { CIScore } from "./types.js";

export class ScoreClient {
  constructor(private client: CommerceIndex) {}

  /** Get full CI Agent Score breakdown. */
  async get(agentId: string): Promise<CIScore> {
    return this.client._request<CIScore>("GET", `/v1/agents/${agentId}/score`);
  }
}
